### PR TITLE
feat: updates to DDA spectral extraction

### PR DESCRIPTION
**Pull request by dmccloskey**
## Description

Minor but important changes to how data-dependent acquisition spectral data is processed using `TargetedSpectralExtractor` so that transitions lists, spectral libraries, and non-targeted spectral library matching can be ran.

## Checklist
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp [hunk 1]
```diff
@@ -748,6 +756,12 @@ namespace OpenMS
 
     // select the best spectrum for each group of spectra having the same name
     selectSpectra(scored_spectra, ms2_features, extracted_spectra, extracted_features, compute_features);
+    
+    // NOTE/DEBUG: commenting out line 758 and uncommenting lines
```
Comment (jpfeuffer) [inline src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp:760]: Not sure if I understand this but maybe enclosing this with an ifdef of a CLASSNAME_DEBUG macro would be better suited than a comment.

File: src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp [hunk 2]
```diff
@@ -266,21 +276,18 @@ namespace OpenMS
...
         }
       }
     }
   }
 
-  void TargetedSpectraExtractor::searchSpectrum(
-      OpenMS::FeatureMap& feat_map,
-      OpenMS::FeatureMap& feat_map_output,
-      bool add_unidentified_features) const
+  void TargetedSpectraExtractor::searchSpectrum(OpenMS::FeatureMap& feat_map, OpenMS::FeatureMap& feat_map_output, bool add_unidentified_features) const
```
Comment (jpfeuffer) [inline src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp:290]: Can't see it from my phone but this change looks like the line could be too long now.

File: src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp [hunk 3]
```diff
@@ -883,54 +898,85 @@ namespace OpenMS
...
+    std::map<std::string, OpenMS::TargetedExperiment::Protein> proteins_map;
     for (const auto& ms1_feature : ms1_features)
     {
       std::string peptide_ref = ms1_feature.getMetaValue("PeptideRef");
       OpenMS::TargetedExperiment::Protein protein;
       protein.id = peptide_ref;
       protein.addMetaValues(ms1_feature);
-      proteins.push_back(protein);
+      auto found_protein = proteins_map.emplace(peptide_ref, protein);
+      if (!found_protein.second)
```
Comment (jpfeuffer) [inline src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp:912]: Is this still needed?

File: src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp [hunk 4]
```diff
@@ -289,6 +296,7 @@ namespace OpenMS
     ams.run(feat_map, output);
     // Remake the feature map replacing the peptide hits as features/sub-features
     feat_map_output.clear();
+    int unknown_feature_counter = 1;
```
Comment (jpfeuffer) [inline src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp:299]: Is this used? Couldn't find it in the added code.

Comment (dmccloskey) [reply src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp:299]: Good catch.  It is removed.

File: src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp (hunk 3 above)
Comment (dmccloskey) [reply src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp:912]: Changed.

File: src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp (hunk 1 above)
Comment (dmccloskey) [reply src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp:760]: I don't think it is needed.  I just added a small note for potential future work.

**Comment by dmccloskey** [comment]
@timosachsenberg and @jpfeuffer I did not see the merge commit.  How can I see what is wrong with the builds on ubuntu/win?  I am looking at the cdash website and I do not see the PR number for any of the builds.  Also, how do I interpret the clang linting?

**Comment by timosachsenberg** [comment]
Hi Douglas, we had an internal migration so a lot of our CI infrastructure had hickups in the last weeks.
The clang lint is currently only suggesting potential style issues - as we can't fully automatize it yet with the current version just ignore it for now. I will restart the build with a branch update. SHould show up on CDASH in an hour or so.

**Comment by dmccloskey** [comment]
> Hi Douglas, we had an internal migration so a lot of our CI infrastructure had hickups in the last weeks. The clang lint is currently only suggesting potential style issues - as we can't fully automatize it yet with the current version just ignore it for now. I will restart the build with a branch update. SHould show up on CDASH in an hour or so.

Thanks.  I can view the Ubuntu errors now.

**Comment by timosachsenberg** [comment]
Yes these are related to some issue with the SQLite qt dependency. Looks good!

**Pull request merged**